### PR TITLE
Fix aria-label and titles for FontAwesomeIcon uses

### PR DESCRIPTION
**Pull request by camd**

### CODE REVIEW

File: ui/job-view/details/summary/SummaryPanel.jsx [hunk 1]
```diff
@@ -56,6 +56,7 @@ class SummaryPanel extends React.PureComponent {
                     icon={faSpinner}
                     pulse
                     className="th-spinner-lg"
+                    aria-label="Loading..."
```
Comment (sarah-clements) [inline ui/job-view/details/summary/SummaryPanel.jsx]: These components appear to have `aria-hidden=true` on by default (looking at the devtools inspector) and so the `aria-label` should go on the parent elements these icons are part of (button, div, span, a, etc).

File: ui/job-view/pushes/RevisionList.jsx [hunk 2]
```diff
@@ -40,7 +40,12 @@ RevisionList.propTypes = {
 export function MoreRevisionsLink(props) {
   return (
     <li>
-      <a href={props.href} target="_blank" rel="noopener noreferrer">
+      <a
+        href={props.href}
+        target="_blank"
+        rel="noopener noreferrer"
+        title="show more revisions"
```
Comment (sarah-clements) [inline ui/job-view/pushes/RevisionList.jsx]: Did you mean to put `title` on the icon?

File: ui/job-view/details/BugFiler.jsx [hunk 3]
```diff
@@ -591,6 +592,7 @@ export class BugFilerClass extends React.Component {
                       <FontAwesomeIcon
                         icon={faExclamationTriangle}
                         id="unhelpful-summary-reason"
+                        aria-hidden
```
Comment (sarah-clements) [inline ui/job-view/details/BugFiler.jsx]: This seems to be true by defaut

**Comment by sarah-clements** [comment]
I was looking over the docs and realized I was looking at the wrong section :) it seems we don't actually need to add the `aria-label` (sorry!). The react-fontawesome library uses SVG with JS, so we just need to add the titles where icons have semantic meaning and it adds the appropriate role and aria label: https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility#svg-semantic

File: ui/job-view/pushes/RevisionList.jsx (hunk 2 above)
Comment (camd) [reply ui/job-view/pushes/RevisionList.jsx]: removed

**Comment by camd** [comment]
I think this is all fixed up now.  As I mentioned in IRC, there are some cases where the parent of the icon has a ``title`` so I didn't add one to the icon.  But I'm curious about your opinion on that.  maybe I should move (or copy) the title to the icon as well?

**Comment by sarah-clements** [comment]
> I think this is all fixed up now. As I mentioned in IRC, there are some cases where the parent of the icon has a `title` so I didn't add one to the icon. But I'm curious about your opinion on that. maybe I should move (or copy) the title to the icon as well?

I think it'd be OK to only have the title in the icon when it's nested in a button (instead of including a `aria-label` on the button). Though I'd be curious how this sounds (or all of TH for that matter!). It seems like an element containing `type="button"` is announced as a button, then "click" and then either the content of the button or the value of an `aria-label` per [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Accessibility_concerns).

**Pull request merged**